runtime/js/98_global_scope.js

```diff
@@ -296,31 +296,15 @@
     Location: location.locationConstructorDescriptor,
     location: location.locationDescriptor,
     Window: globalInterfaces.windowConstructorDescriptor,
-    window: util.readOnly(globalThis),
-    self: util.writable(globalThis),
+    window: util.getterOnly(() => globalThis),
+    self: util.getterOnly(() => globalThis),
     Navigator: util.nonEnumerable(Navigator),
-    navigator: {
-      configurable: true,
-      enumerable: true,
-      get: () => navigator,
-    },
+    navigator: util.getterOnly(() => navigator),
     alert: util.writable(prompt.alert),
     confirm: util.writable(prompt.confirm),
     prompt: util.writable(prompt.prompt),
-    localStorage: {
-      configurable: true,
-      enumerable: true,
-      get: webStorage.localStorage,
-      // Makes this reassignable to make astro work
-      set: () => {},
-    },
```

Comment:
I don't like this change, it will break Astro again

Reply:
@bartlomieju it doesnt. its changed to `getterOnly` with does this.